### PR TITLE
Finishing mvp

### PR DESCRIPTION
**Pull request by javiguerra777**
## Changes
1. Finished the styling for the UI
2. Functionality of the game is complete and connected to a database
3. Finished the redux store for registration

## Purpose
This feature is to finish adding the last tad bit of functionality needed to finish the MVP of this web application
## Approach
This allows the app to be ready for deployment and production.
## Learning
I learned how to connect the phaser game to a backend that uses websockets or socket.io. I used this article as reference

https://phaser.io/news/2017/03/socketio-multiplayer-tutorial

### CODE REVIEW

File: src/store/UserSlice.ts [hunk 1]
```diff
@@ -14,6 +15,7 @@ const initialState = {
   username: '',
   name: '',
   email: '',
+  host: false,
```
Comment (srirachy) [inline src/store/UserSlice.ts:18]: did this fix the host thing?

File: src/styles/AnimSprite.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  @keyframes moveSpritesheet {
+    from {
+      transform: translate3d(0px, 0, 0);
+    }
+    to {
+      transform: translate3d(-100%, 0, 0);
+    }
+  }
+`;
+
```
Comment (srirachy) [inline src/styles/AnimSprite.ts:53]: much styles

File: src/utils/api.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+};
+export const getRoomData = async (id: string) => {
+  const response = await axios.get(`${dbUrl}/room/${id}`);
+  return response;
+};
+export const getAllRooms = async () => {
+  const response = await axios.get(`${dbUrl}/rooms`);
+  return response;
+};
+
```
Comment (srirachy) [inline src/utils/api.ts:21]: damn you re-used the shit outta dbUrl

File: src/phaser-game/scenes/HomeScene.ts [hunk 4]
```diff
@@ -293,34 +335,16 @@ class HomeScene extends Phaser.Scene {
...
-        this.homeOtherPlayer.anims.play('rightStill', true);
-      } else if (this.homeOtherPlayer.direction === 'left') {
-        this.homeOtherPlayer.anims.play('leftStill', true);
-      } else if (this.homeOtherPlayer.direction === 'up') {
-        this.homeOtherPlayer.anims.play('upStill', true);
-      } else if (this.homeOtherPlayer.direction === 'down') {
-        this.homeOtherPlayer.anims.play('downStill', true);
-      }
+    if (this.homeOtherPlayer) {
+      this.handleOtherPlayerAnims();
```
Comment (srirachy) [inline src/phaser-game/scenes/HomeScene.ts:347]: damn all this code got handled

File: src/phaser-game/scenes/HomeScene.ts [hunk 5]
```diff
@@ -228,10 +259,11 @@ class HomeScene extends Phaser.Scene {
     socket.on('existingPlayer', async (data) => {
       try {
         this.homeOtherPlayer = this.physics.add.sprite(
-          500,
-          500,
+          data.x,
+          data.y,
```
Comment (srirachy) [inline src/phaser-game/scenes/HomeScene.ts:263]: nice

File: src/pages/Game.tsx [hunk 6]
```diff
@@ -161,9 +205,9 @@ function Game() {
             <button type="button" onClick={asideOptions}>
               <AiOutlineWechat />
             </button>
-            <button type="button">
+            <button type="button" onClick={displayUsers}>
               <BsPeople />
```
Comment (srirachy) [inline src/pages/Game.tsx:209]: bs people

File: src/pages/Game.tsx [hunk 7]
```diff
@@ -57,13 +78,29 @@ function Game() {
...
+      try {
+        getRoomData(id || '').then((res) => setRoomData(res.data));
+      } catch (err) {
+        if (err instanceof Error) {
+          setMessage(err.message);
+          navigate('/dashboard');
+        }
+      }
+    };
+    resolveRoom();
```
Comment (srirachy) [inline src/pages/Game.tsx:92]: damn this room do be resolved

File: src/pages/HomePage.tsx [hunk 8]
```diff
@@ -141,13 +149,10 @@ function HomePage() {
             src={dashboardimage}
             alt="dashboard"
           />
-          <iframe
-            className="prev-vid"
-            src="https://www.youtube.com/embed/T33NN_pPeNI"
```
Comment (srirachy) [inline src/pages/HomePage.tsx:146]: i checked the vid out and liked and subscribed

File: src/phaser-game/scenes/FortNerf.ts [hunk 9]
```diff
@@ -198,7 +198,6 @@ class FortNerf extends Phaser.Scene {
       }
       this.playerText.setX(this.player.x - 30);
       this.playerText.setY(this.player.y + 30);
-
```
Comment (srirachy) [inline src/phaser-game/scenes/FortNerf.ts:201]: huge

**Review by srirachy — APPROVED** [review]
good stuff

File: src/store/UserSlice.ts (hunk 1 above)
Comment (javiguerra777) [reply src/store/UserSlice.ts:18]: no, it did not. i decided to scrape the host thing for now

File: src/pages/HomePage.tsx (hunk 8 above)
Comment (javiguerra777) [reply src/pages/HomePage.tsx:146]: haha funny, i was using it as placeholder until i created my own vid

**Pull request merged**